components/aws/sagemaker/tests/integration_tests/component_tests/test_v2_monitoring_schedule_component.py

```diff
@@ -0,0 +1,258 @@
+
+    # parameters for job definition
+    test_params["Arguments"][test_params["JobInputName"]]["endpointInput"][
+        "endpointName"
+    ] = deploy_endpoint
+
+    job_definition_name_1 = (
+        utils.generate_random_string(10) + "-v2-model-data-quality-defi"
+    )
+    job_definition_name_2 = (
+        utils.generate_random_string(10) + "-v2-model-data-quality-defi"
+    )
+
+    # parameter for monitoring schedule
+    monitoring_schedule_name = (
+        utils.generate_random_string(10) + "-v2-monitoring-schedule"
+    )
+
+    try:
+        test_params["Arguments"]["job_definition_name"] = job_definition_name_1
+        test_params["Arguments"]["job_resources"]["clusterConfig"][
+            "instanceType"
+        ] = "ml.m5.large"
+        test_params["Arguments"]["monitoring_schedule_name"] = monitoring_schedule_name
+
+        _, _, _ = kfp_client_utils.compile_run_monitor_pipeline(
+            kfp_client,
+            experiment_id,
+            test_params["PipelineDefinition"],
+            test_params["Arguments"],
+            download_dir,
+            test_params["TestName"],
+            test_params["Timeout"],
+        )
+
+        job_definition_1_describe_ack = ack_utils._get_resource(
+            k8s_client, job_definition_name_1, test_params["Plural"]
+        )
+
+        # Check if the job definition is created
+        assert (
+            job_definition_1_describe_ack["status"]["ackResourceMetadata"]["arn"]
+            != None
+        )
+
+        # Verify if monitoring schedule is created with correct name and endpoint
+        monitoring_schedule_describe = sagemaker_client.describe_monitoring_schedule(
+            MonitoringScheduleName=monitoring_schedule_name
+        )
+        assert (
+            monitoring_schedule_name
+            in monitoring_schedule_describe["MonitoringScheduleArn"]
+        )
+
+        assert monitoring_schedule_describe["MonitoringScheduleStatus"] == "Scheduled"
+
+        assert monitoring_schedule_describe["EndpointName"] == deploy_endpoint
+
+        assert (
+            monitoring_schedule_describe["MonitoringScheduleConfig"][
+                "MonitoringJobDefinitionName"
+            ]
+            == job_definition_name_1
+        )
+
+        # Verify if job definition is created with correct instance type
+        job_definition_1_describe = sagemaker_client.describe_data_quality_job_definition(
+            JobDefinitionName=job_definition_name_1
+        )
+        assert (
+            job_definition_1_describe["JobResources"]["ClusterConfig"]["InstanceType"]
+            == "ml.m5.large"
+        )
+
+        # Update monitoring schedule using new job definition
+        test_params["Arguments"]["job_definition_name"] = job_definition_name_2
+        test_params["Arguments"]["job_resources"]["clusterConfig"][
+            "instanceType"
+        ] = "ml.m5.xlarge"
+
+        _, _, _ = kfp_client_utils.compile_run_monitor_pipeline(
```

Comment:
not running same pipeline but this is good enough for coverage

Reply:
It's the same pipeline but with different input parameters (job_definition_name and instance_type). We explicitly specify job_definition_name instead of letting the component auto configure the name, because it is easier to check output and delete resources.